src/base/AsyncCallList.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_BASE_ASYNCCALLLIST_H
+#define SQUID_BASE_ASYNCCALLLIST_H
+
+#include "base/forward.h"
+#include "base/RefCount.h"
+
+/// An efficient (but intrusive) AsyncCall storage preserving FIFO order.
+/// A given AsyncCall object may reside in at most one such storage.
+class AsyncCallList
```

Comment:
> The documentation you are reading now applies to the (duplicated) code you have approved during the previous iteration

To clarify, the documentation applies to old/official AsyncCallQueue code from where this implementation was moved into a dedicated class (AsyncCallList) and to old/official Comm uses of the same invasive list for close handlers. The delay readers code simplified in this PR used an std::vector before ff0ac21 and then was converted to use the same AsyncCallList class to avoid reimplementing that code functionality and to gain AsyncCallList efficiencies.

Reply:
Ok. let's keep the old intrusive-list implementation and revisit this area in a future improvement.